core/corehttp/routing.go

```diff
@@ -0,0 +1,129 @@
+package corehttp
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/ipfs/boxo/ipns"
+	"github.com/ipfs/boxo/routing/http/server"
+	"github.com/ipfs/boxo/routing/http/types"
+	"github.com/ipfs/boxo/routing/http/types/iter"
+	cid "github.com/ipfs/go-cid"
+	core "github.com/ipfs/kubo/core"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+)
+
+func RoutingOption() ServeOption {
+	return func(n *core.IpfsNode, _ net.Listener, mux *http.ServeMux) (*http.ServeMux, error) {
+		handler := server.Handler(&contentRouter{n})
+		mux.Handle("/routing/v1/", handler)
+		return mux, nil
+	}
+}
+
+type contentRouter struct {
+	n *core.IpfsNode
+}
+
+func (r *contentRouter) FindProviders(ctx context.Context, key cid.Cid, limit int) (iter.ResultIter[types.Record], error) {
+	ctx, cancel := context.WithCancel(ctx)
+	ch := r.n.Routing.FindProvidersAsync(ctx, key, limit)
+	return iter.ToResultIter[types.Record](&peerChanIter{
+		ch:     ch,
+		cancel: cancel,
+	}), nil
+}
+
+// nolint deprecated
+func (r *contentRouter) ProvideBitswap(ctx context.Context, req *server.BitswapWriteProvideRequest) (time.Duration, error) {
+	return 0, routing.ErrNotSupported
+}
+
+func (r *contentRouter) FindPeers(ctx context.Context, pid peer.ID, limit int) (iter.ResultIter[types.Record], error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	addr, err := r.n.Routing.FindPeer(ctx, pid)
+	if err != nil {
+		return nil, err
+	}
+
+	rec := &types.PeerRecord{
+		Schema: types.SchemaPeer,
+		ID:     &addr.ID,
+	}
+
+	for _, addr := range addr.Addrs {
+		rec.Addrs = append(rec.Addrs, types.Multiaddr{Multiaddr: addr})
+	}
+
+	return iter.ToResultIter[types.Record](iter.FromSlice[types.Record]([]types.Record{rec})), nil
+}
+
+func (r *contentRouter) GetIPNS(ctx context.Context, name ipns.Name) (*ipns.Record, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	raw, err := r.n.Routing.GetValue(ctx, string(name.RoutingKey()))
+	if err != nil {
+		return nil, err
+	}
+
+	return ipns.UnmarshalRecord(raw)
+}
+
+func (r *contentRouter) PutIPNS(ctx context.Context, name ipns.Name, record *ipns.Record) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	raw, err := ipns.MarshalRecord(record)
+	if err != nil {
+		return err
+	}
+
+	// The caller guarantees that name matches the record. This is double checked
+	// by the internals of PutValue.
+	return r.n.Routing.PutValue(ctx, string(name.RoutingKey()), raw)
```

Comment:
@hacdias what exactly happens inside PutValue? 

kad-dht's `PutValue` calls IPNS record validation here:
https://github.com/libp2p/go-libp2p-kad-dht/blob/master/routing.go#L45-L47

is this the only check we do? 


iiuc if someone uses a different router, they just make a blind put of a potentially invalid record. 

Could we add an explicit IPNS Record validation inside `PutIPNS` here, before passing to the routing system via `PutValue`?

Reply:
@lidel I would argue whoever implements `PutValue` is the one responsible for validating it. Our `boxo/routing/http` does this too. Because there may also be other types of values you want to support.